crates/tap-agent/src/test.rs

```diff
@@ -31,6 +58,193 @@ lazy_static! {
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(
+                network_subgraph_endpoint.unwrap_or(&get_grpc_url().await),
+            )
+            .unwrap(),
+        )
+        .await,
+    ));
+    let escrow_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(
+                escrow_subgraph_endpoint.unwrap_or(&get_grpc_url().await),
+            )
+            .unwrap(),
+        )
+        .await,
+    ));
+    let (escrow_accounts_tx, escrow_accounts_rx) = watch::channel(EscrowAccounts::default());
+    escrow_accounts_tx
+        .send(EscrowAccounts::new(
+            HashMap::from([(SENDER.1, U256::from(ESCROW_VALUE))]),
+            HashMap::from([(SENDER.1, vec![SIGNER.1])]),
+        ))
+        .expect("Failed to update escrow_accounts channel");
+
+    let prefix = format!(
+        "test-{}",
+        PREFIX_ID.fetch_add(1, std::sync::atomic::Ordering::SeqCst)
+    );
+
+    let args = SenderAccountArgs {
+        config,
+        pgpool,
+        sender_id: SENDER.1,
+        escrow_accounts: escrow_accounts_rx,
+        indexer_allocations: watch::channel(initial_allocation).1,
+        escrow_subgraph,
+        network_subgraph,
+        domain_separator: TAP_EIP712_DOMAIN_SEPARATOR.clone(),
+        sender_aggregator_endpoint: aggregator_endpoint
+            .unwrap_or(Url::parse(&get_grpc_url().await).unwrap()),
+        allocation_ids: HashSet::new(),
+        prefix: Some(prefix.clone()),
+        retry_interval: RETRY_DURATION,
+    };
+
+    let actor = TestableActor::new(SenderAccount);
+    let notify = actor.notify.clone();
+
+    let (sender, _) = Actor::spawn(Some(prefix.clone()), actor, args)
+        .await
+        .unwrap();
+
+    // flush all messages
+    flush_messages(&notify).await;
+
+    (sender, notify, prefix, escrow_accounts_tx)
+}
+
+#[bon::builder]
+pub async fn create_sender_accounts_manager(
+    pgpool: PgPool,
+    aggregator_endpoint: Option<Url>,
+    network_subgraph: Option<&str>,
+    escrow_subgraph: Option<&str>,
+) -> (
+    String,
+    Arc<Notify>,
+    (ActorRef<SenderAccountsManagerMessage>, JoinHandle<()>),
+) {
+    let config = get_sender_account_config();
+    let (_allocations_tx, allocations_rx) = watch::channel(HashMap::new());
+    let escrow_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(escrow_subgraph.unwrap_or(&get_grpc_url().await))
+                .unwrap(),
+        )
+        .await,
+    ));
+    let network_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(network_subgraph.unwrap_or(&get_grpc_url().await))
+                .unwrap(),
+        )
```

Comment:
`get_grpc_url()` should only be used for aggregator, I think a DUMMY_URL works here.